dev/ScrollViewer/ScrollViewer.cpp

```diff
@@ -1555,6 +1671,172 @@ void ScrollViewer::HideIndicatorsAfterDelay()
     }
 }
 
+// On RS4 and RS5, update m_autoHideScrollControllers based on the DynamicScrollbars registry key value
+// and update the visual states if the value changed.
+void ScrollViewer::UpdateScrollControllersAutoHiding(
+    bool forceUpdate)
+{
+    if ((forceUpdate || (!m_uiSettings5 && SharedHelpers::IsRS4OrHigher())) && m_autoHideScrollControllersValid)
+    {
+        m_autoHideScrollControllersValid = false;
+
+        bool oldAutoHideScrollControllers = m_autoHideScrollControllers;
+        bool newAutoHideScrollControllers = AreScrollControllersAutoHiding();
+
+        if (oldAutoHideScrollControllers != newAutoHideScrollControllers)
+        {
+            UpdateVisualStates(
+                true  /*useTransitions*/,
+                false /*showIndicators*/,
+                false /*hideIndicators*/,
+                true  /*scrollControllersAutoHidingChanged*/);
+        }
+    }
+}
+
+void ScrollViewer::UpdateVisualStates(
+    bool useTransitions,
+    bool showIndicators,
+    bool hideIndicators,
+    bool scrollControllersAutoHidingChanged,
+    bool updateScrollControllersAutoHiding,
+    bool onlyForAutoHidingScrollControllers)
+{
+    if (updateScrollControllersAutoHiding)
+    {
+        UpdateScrollControllersAutoHiding();
+    }
+
+    if (onlyForAutoHidingScrollControllers && !AreScrollControllersAutoHiding())
+    {
+        return;
+    }
+
+    UpdateScrollControllersVisualState(useTransitions, showIndicators, hideIndicators);
+    UpdateScrollControllersSeparatorVisualState(useTransitions, scrollControllersAutoHidingChanged);
+}
+
+// Updates the state for the ScrollingIndicatorStates state group.
+void ScrollViewer::UpdateScrollControllersVisualState(
+    bool useTransitions,
+    bool showIndicators,
+    bool hideIndicators)
+{
+    SCROLLVIEWER_TRACE_VERBOSE(*this, TRACE_MSG_METH_INT, METH_NAME, this, useTransitions);
+    SCROLLVIEWER_TRACE_VERBOSE(*this, TRACE_MSG_METH_INT_INT, METH_NAME, this, showIndicators, hideIndicators);
+
+    MUX_ASSERT(!(showIndicators && hideIndicators));
+
+    bool areScrollControllersAutoHiding = AreScrollControllersAutoHiding();
+
+    MUX_ASSERT(!(!areScrollControllersAutoHiding && hideIndicators));
+
+    if ((!areScrollControllersAutoHiding || showIndicators) && !hideIndicators)
+    {
+        if (AreAllScrollControllersCollapsed())
+        {
+            return;
+        }
+
+        ResetHideIndicatorsTimer(false /*isForDestructor*/, true /*restart*/);
+
+        // Mouse indicators dominate if they are already showing or if we have set the flag to prefer them.
+        if (m_preferMouseIndicators || m_showingMouseIndicators || !areScrollControllersAutoHiding)
+        {
+            GoToState(s_mouseIndicatorStateName, useTransitions);
+
+            m_showingMouseIndicators = true;
```

Comment:
When is this set back to false? This method is kind of a hodgepodge of responsibilities.

Reply:
ScrollViewer::OnScrollViewerUnloaded, ScrollViewer::HideIndicators and ScrollViewer::OnNoIndicatorStateStoryboardCompleted set it back to false. Don't think this is worth changing at this point.